### PR TITLE
feature: Add ask to repeat TrainingProgress status

### PR DESCRIPTION
**Pull request by lauryndbrown**
Fixes https://github.com/carpentries/amy/issues/1808

Adds the new column and moves all those who currently have "Failed" status to "Asked to Repeat". This also updates the fake data script.  
Screen Shot 2021-03-02 at 6 54 11 AM

TODO: 
- [ ] Fix tests
- [ ] Determine if the failed status behavior needs to change (i.e. should it prevent a trainee from enrolling again? Does it already?)

### CODE REVIEW

**Comment by lauryndbrown** [comment]
@maneesha I'm a little curious about the behavior of ask to repeat vs failed. 

For Ask to repeat looks like:

Screen Shot 2021-03-02 at 7 09 40 AM

And Failed looks like:
Screen Shot 2021-03-02 at 7 11 10 AM

The others are the same. Would we prefer a different color used? Also do we want failed status to provide any additional behavior than it used to?

**Comment by maneesha** [comment]
@lauryndbrown Part of The Carpentries' philosophy is to establish a culture of opportunity and growth.  That's why we do not want to fail people; rather we want to give them the opportunity to try again.  We do want to reserve the "fail" option for extremely rare instances where someone does something so inappropriate that we do not want to ask them to try again. You bring up a good point about whether the "fail" should prevent them from enrolling again.  I'll check with our team and get back to you.  

Using colors green for pass; yellow for ask to repeat; and red for fail works.

**Comment by maneesha** [comment]
@lauryndbrown
I left this note in the associated issue, but realize I should have left it here instead.  

> If someone has a "failed" status for a Training Progress step, then if we try to award them a badge or try to enroll them in a new instructor training event, we should get a warning saying they failed this step and asking if we are sure we want to proceed.
> 
> Also if someone has a "failed" status, Notes should be required. Notes are not required for any other status.

**Comment by lauryndbrown** [comment]
Notes required in failed state:

Screen Shot 2021-03-06 at 7 46 24 PM

**Comment by lauryndbrown** [comment]
Failed state requires notes:
Screen Shot 2021-03-06 at 7 46 24 PM

**Comment by lauryndbrown** [comment]
Failed state requires notes:
Screen Shot 2021-03-06 at 7 49 55 PM

**Comment by lauryndbrown** [comment]
Warning when creating new award:
Screen Shot 2021-03-06 at 8 57 04 PM

**Comment by lauryndbrown** [comment]
Warning when creating new award (from person edit):
Screen Shot 2021-03-06 at 8 58 25 PM

**Comment by maneesha** [comment]
@lauryndbrown  Checking in - Is there anything else you need from me to complete this?

**Comment by lauryndbrown** [comment]
@maneesha I don't believe so, I think this ended up delayed when I started on consents. I'll revise this tomorrow put it up for review

**Comment by lauryndbrown** [comment]
@maneesha Ok changed the way the admin is alerted that the trainee failed a training using a popup to keep things more consistent:

Person edit view:
Screen Shot 2021-04-26 at 9 08 44 AM


Screen Shot 2021-04-26 at 9 02 59 AM

Event edit view:
Screen Shot 2021-04-26 at 9 06 34 AM

File: amy/templates/workshops/event_edit_form.html [hunk 1]
```diff
@@ -75,5 +75,16 @@
   $(function() {
     $('#tabs').stickyTabs();
   });
+  $('select[name="task-person"]').on('change', function() {
+    var person_id = $(this).val();
+    var url = `/api/v1/persons/${person_id}/trainingprogress`;
+    $.get(url, function (training_progresses) {   
+      for (training of training_progresses){
+        if (training.state === "Failed"){
+          alert("Warning: Trainee failed previous training(s). Are you sure you want to continue?");
```
Comment (pbanaszkiewicz) [inline amy/templates/workshops/event_edit_form.html:84]: Maybe add trainee name here? Otherwise user may face n alerts with no additional information.

**Review by pbanaszkiewicz — APPROVED** [review]
I left minor suggestion, but it looks great nonetheless.

**Pull request merged**